pkg/shp/cmd/build/upload.go

```diff
@@ -126,23 +159,36 @@ func (u *UploadCommand) Validate() error {
 
 // createBuildRun creates the BuildRun instance to receive the data upload afterwards, it returns the
 // BuildRun name just created and error.
-func (u *UploadCommand) createBuildRun(p *params.Params) (*types.NamespacedName, error) {
-	buildRefName := u.buildRunSpec.BuildRef.Name
-	u.buildRunSpec.Sources = &[]buildv1alpha1.BuildSource{{
-		Name: "local-copy",
-		Type: buildv1alpha1.LocalCopy,
-	}}
-	br := &buildv1alpha1.BuildRun{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: fmt.Sprintf("%s-", buildRefName),
-		},
-		Spec: *u.buildRunSpec,
+func (u *UploadCommand) createBuildRun(p *params.Params) (*buildv1alpha1.BuildRun, error) {
```

Comment:
Following the suggestion of only using `u.sourceBundleImage`, here it would become a just if/else block.